tls/ttls.c

```diff
@@ -2218,6 +2231,11 @@ ttls_recv(void *tls_data, unsigned char *buf, size_t len, unsigned int *read)
 		if (unlikely(!ttls_xfrm_ready(tls))) {
 			if (!(r = ttls_handle_alert(io)))
 				goto skip_record;
+			if (io->alert[0] == TTLS_ALERT_LEVEL_WARNING
+			    && io->alert[1] == TTLS_ALERT_MSG_CLOSE_NOTIFY)
+			{
+				ttls_close_notify(tls);
+			}
```

Comment:
`ttls_recv()` is overloaded and you copy & paste the same code, so please more the code to `ttls_handle_alert()` and amend the function to accept `tls` instead of `io`.